internal/armada/metrics/domain_test.go

```diff
@@ -0,0 +1,95 @@
+package metrics
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	armadaresource "github.com/armadaproject/armada/internal/common/resource"
+)
+
+func TestRecordJobRuntime_ShouldReportRuntimeDurations(t *testing.T) {
+	jobMetricsRecorder := NewJobMetricsRecorder()
+	jobMetricsRecorder.RecordJobRuntime("cpu", "", time.Second)
+	jobMetricsRecorder.RecordJobRuntime("cpu", "", 2*time.Second)
+	jobMetricsRecorder.RecordJobRuntime("cpu", "", 3*time.Second)
+
+	jobMetricsRecorder.RecordJobRuntime("gpu", "", 2*time.Second)
```

Comment:
I don't understand the labels "cpu" with duration.  It seems weird to me..

Reply:
yes they can serve as arbitrary pool tag names, but I think when I wrote them I misunderstood the context here. 

> I wonder if this test should also verify with having a couple different priority classes for the same pool?

I believe it will create just another `QueueMetricsRecorder`, for example if we recorded `pool1` with priority classes `1` and `2`, we will have 2 `QueueMetricsRecorder` with different `Metrics`
https://github.com/armadaproject/armada/blob/371fd974cb02ae70fb5c4a54cdedfe8ed487b780/internal/armada/metrics/domain.go#L55-L67

you think we should verify that both metrics are calculated separately ?